tests/anisotropic_viscosity.cc

```diff
@@ -0,0 +1,161 @@
+        {
+          const double pressure = in.pressure[i];
+          out.densities[i] = 1.0 + pressure;
+          out.compressibilities[i] = 1.0 / (1. + pressure);
+          if ((in.position[i]-center).norm() < 0.25)
+            {
+              out.constitutive_tensors[i] = C;
+            }
+        }
+    }
+
+    template <int dim>
+    bool
+    Anisotropic<dim>::
+    is_compressible () const
+    {
+      return true;
+    }
+
+    template <int dim>
+    void
+    Anisotropic<dim>::
+    declare_parameters (ParameterHandler &prm)
+    {
+      Simple<dim>::declare_parameters (prm);
+      prm.enter_subsection ("Material model");
+      {
+        prm.enter_subsection ("Anisotropic");
+        {
+          if (dim == 2)
+            prm.declare_entry ("Viscosity tensor",
+                               "1, 0, 0,"
+                               "0, 1, 0,"
+                               "0, 0,.5",
+                               Patterns::List(Patterns::Double()),
+                               "Viscosity-scaling tensor in Voigt notation.");
+          else
+            prm.declare_entry ("Viscosity tensor",
+                               "1, 0, 0, 0, 0, 0,"
+                               "0, 1, 0, 0, 0, 0,"
+                               "0, 0, 1, 0, 0, 0,"
+                               "0, 0, 0,.5, 0, 0,"
+                               "0, 0, 0, 0,.5, 0,"
+                               "0, 0, 0, 0, 0,.5",
+                               Patterns::List(Patterns::Double()),
+                               "Viscosity-scaling tensor in Voigt notation.");
+        }
+        prm.leave_subsection();
+      }
+      prm.leave_subsection();
+    }
+
+    template <int dim>
+    void
+    Anisotropic<dim>::
+    parse_parameters (ParameterHandler &prm)
+    {
+      Simple<dim>::parse_parameters (prm);
+      prm.enter_subsection ("Material model");
+      {
+        prm.enter_subsection ("Anisotropic");
+        {
+          const int size_voigt = (dim == 3 ? 6 : 3);
+          const std::vector<double> tmp_tensor =
+            Utilities::string_to_double(Utilities::split_string_list(prm.get ("Viscosity tensor")));
+          Assert(tmp_tensor.size() == size_voigt*size_voigt,
+                 ExcMessage("Constitutive voigt matrix must have 9 components in 2D, or 36 components in 3d"));
+
+          std::vector<std::vector<double> > voigt_visc_tensor (size_voigt);
+          for (unsigned int i=0; i<size_voigt; ++i)
+            {
+              voigt_visc_tensor[i].resize(size_voigt);
+              for (unsigned int j=0; j<size_voigt; ++j)
+                voigt_visc_tensor[i][j] = tmp_tensor[i*size_voigt+j];
+            }
+
+          // Voigt indices (For mapping back to real tensor)
+          const unsigned int vi3d0[] = {0, 1, 2, 1, 0, 0};
+          const unsigned int vi3d1[] = {0, 1, 2, 2, 2, 1};
+          const unsigned int vi2d0[] = {0, 1, 0};
+          const unsigned int vi2d1[] = {0, 1, 1};
```

Comment:
Ah, cool -- I think I always wanted to have a function that translates between Voigt and symmetric-rank-4 notation. Now I know where to look for.